### ISSUE TITLE
video_player_web,video_player can play in Mac safari normaly,but when I play in phone safari, it doesn't work

### ISSUE THREAD

**Issue by qiuguian** [flutter/flutter#47030]
video_player_web,video_player can play in Mac safari normaly by platform web, but when I play in phone safari, it doesn't work.

**Comment by qiuguian** [comment]
https://file-examples.com/wp-content/uploads/2017/04/file_example_MP4_480_1_5MG.mp4;

**Comment by qiuguian** [comment]
I  use video_player can play in Mac safari . when play in phone safari or other android phone, it doesn't work!

**Comment by qiuguian** [comment]
![image](https://user-images.githubusercontent.com/7962840/70846003-f81c9200-1e8f-11ea-8861-e8aa2f41d0f3.png)

**Comment by qiuguian** [comment]
![WechatIMG365](https://user-images.githubusercontent.com/7962840/70846104-10d97780-1e91-11ea-92e5-b900863bb134.jpeg)

**Comment by qiuguian** [comment]
play in iphone  <img width="868" alt="截屏2019-12-1420 00 42" src="https://user-images.githubusercontent.com/7962840/70848445-9fa7bd80-1eac-11ea-82f7-f49acae513df.png">

**Comment by iapicca** [comment]
Hi @qiuguian 
can you please provide your updated  `flutter doctor -v`
and your `flutter run --verbose`?
Also, to better address the issue, would be helpful
if you could post  a minimal code sample
or the steps to reproduce the problem.

Please refrain from posting pictures if not necessary.

Thank you

**Comment by qiuguian** [comment]
Thank you for respose!, you can try video_player  example,  first flutter build web, and them play in your phone broser, for iPhone safari, or other android device.

flutter:channel master v1.13.1-pre.93
device: iPhone 8 Plus
tool: Android Studio

![image](https://user-images.githubusercontent.com/7962840/70893329-d8b76d80-2025-11ea-806e-a848b59c3f24.png)

import 'package:video_player/video_player.dart';
import 'package:flutter/material.dart';

here is  the code ,it can player in Mac safari, when in phone ,it can't!

void main() => runApp(VideoApp());

class VideoApp extends StatefulWidget {
  @override
  _VideoAppState createState() => _VideoAppState();
}

class _VideoAppState extends State<VideoApp> {
  VideoPlayerController _controller;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network(
        'http://www.sample-videos.com/video123/mp4/720/big_buck_bunny_720p_20mb.mp4')
      ..initialize().then((_) {
        // Ensure the first frame is shown after the video is initialized, even before the play button has been pressed.
        setState(() {});
      });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Video Demo',
      home: Scaffold(
        body: Center(
          child: _controller.value.initialized
              ? AspectRatio(
                  aspectRatio: _controller.value.aspectRatio,
                  child: VideoPlayer(_controller),
                )
              : Container(),
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: () {
            setState(() {
              _controller.value.isPlaying
                  ? _controller.pause()
                  : _controller.play();
            });
          },
          child: Icon(
            _controller.value.isPlaying ? Icons.pause : Icons.play_arrow,
          ),
        ),
      ),
    );
  }

  @override
  void dispose() {
    super.dispose();
    _controller.dispose();
  }
}

**Comment by iapicca** [comment]
Hi @qiuguian 
could you please provide your `flutter run --verbose`?

Please refrain from posting pictures if not necessary.

thank you

**Comment by qiuguian** [comment]
bogon:video_client_001 qiuguian$ flutter run --verbose
[  +47 ms] executing: [/Users/qiuguian/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +87 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 01953a101576a64aeaec48d434e780ce31e50538
[        ] executing: [/Users/qiuguian/flutter/] git describe --match v*.*.* --first-parent --long --tags
[ +131 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[   +4 ms] v1.13.0-93-g01953a101
[  +18 ms] executing: [/Users/qiuguian/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/Users/qiuguian/flutter/] git ls-remote --get-url origin
[  +28 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +133 ms] executing: [/Users/qiuguian/flutter/] git rev-parse --abbrev-ref HEAD
[  +31 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[  +52 ms] master
[  +30 ms] executing: sw_vers -productName
[  +32 ms] Exit code 0 from: sw_vers -productName
[        ] Mac OS X
[        ] executing: sw_vers -productVersion
[  +37 ms] Exit code 0 from: sw_vers -productVersion
[        ] 10.15.1
[        ] executing: sw_vers -buildVersion
[  +29 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 19B88
[  +50 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +87 ms] executing: /Users/qiuguian/Library/Android/sdk/platform-tools/adb devices -l
[  +20 ms] Exit code 0 from: /Users/qiuguian/Library/Android/sdk/platform-tools/adb devices -l
[        ] List of devices attached
[  +25 ms] executing: /Users/qiuguian/flutter/bin/cache/artifacts/libimobiledevice/idevice_id -h
[ +131 ms] Error getting attached iOS device: ideviceinfo could not find device:
                    No device found with udid 04cf4709310d0462e97e2b219c89c7f73acfc4f2, is it plugged in?
                    . Try unlocking attached devices.
[   +4 ms] executing: /usr/bin/xcode-select --print-path
[  +10 ms] Exit code 0 from: /usr/bin/xcode-select --print-path
[        ] /Applications/Xcode.app/Contents/Developer
[   +4 ms] executing: /usr/bin/xcodebuild -version
[ +852 ms] Exit code 0 from: /usr/bin/xcodebuild -version
[        ] Xcode 11.2.1
           Build version 11B500
[   +4 ms] /usr/bin/xcrun simctl list --json devices
[ +237 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +102 ms] More than one device connected; please specify a device with the '-d <deviceId>' flag, or use '-d all' to act on all devices.
[ +187 ms] Chrome     • chrome     • web-javascript • Google Chrome 78.0.3904.108
[   +1 ms] Web Server • web-server • web-javascript • Flutter Tools
[  +19 ms] "flutter run" took 1,971ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:302:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:601:11)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:516:33)
<asynchronous suspension>
#4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#5      _rootRun (dart:async/zone.dart:1126:13)
#6      _CustomZone.run (dart:async/zone.dart:1023:19)
#7      _runZoned (dart:async/zone.dart:1518:10)
#8      runZoned (dart:async/zone.dart:1465:12)
#9      AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#10     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:506:20)
#11     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#12     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:333:21)
<asynchronous suspension>
#13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#14     _rootRun (dart:async/zone.dart:1126:13)
#15     _CustomZone.run (dart:async/zone.dart:1023:19)
#16     _runZoned (dart:async/zone.dart:1518:10)
#17     runZoned (dart:async/zone.dart:1465:12)
#18     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#19     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:284:19)
#20     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#21     new Future.sync (dart:async/future.dart:224:31)
#22     CommandRunner.run (package:args/command_runner.dart:112:14)
#23     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:227:18)
#24     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:22)
#25     _rootRun (dart:async/zone.dart:1126:13)
#26     _CustomZone.run (dart:async/zone.dart:1023:19)
#27     _runZoned (dart:async/zone.dart:1518:10)
#28     runZoned (dart:async/zone.dart:1502:12)
#29     run.<anonymous closure> (package:flutter_tools/runner.dart:62:18)
<asynchronous suspension>
#30     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#31     _rootRun (dart:async/zone.dart:1126:13)
#32     _CustomZone.run (dart:async/zone.dart:1023:19)
#33     _runZoned (dart:async/zone.dart:1518:10)
#34     runZoned (dart:async/zone.dart:1465:12)
#35     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#36     runInContext (package:flutter_tools/src/context_runner.dart:64:24)
#37     run (package:flutter_tools/runner.dart:51:10)
#38     main (package:flutter_tools/executable.dart:65:9)
#39     main (file:///Users/qiuguian/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#40     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:305:32)
#41     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)



bogon:video_client_001 qiuguian$

**Comment by iapicca** [comment]
Hi @qiuguian 
are those logs from `video_player `running on an iphone's safari, correct?
thank you

**Comment by qiuguian** [comment]
flutter build web, just the video_player can't play on phone,but  on Mac's  safari, is all correct.

**Comment by qiuguian** [comment]
I also try the futter_galley example, just like this.
![image](https://user-images.githubusercontent.com/7962840/70895036-24b7e180-2029-11ea-952c-55afa1fb8c7c.png)

**Comment by qiuguian** [comment]
the video_player can't initialize()

**Comment by qiuguian** [comment]
corrent on Mac

<img width="496" alt="截屏2019-12-1617 34 59" src="https://user-images.githubusercontent.com/7962840/70895969-d0adfc80-202a-11ea-80a4-795a1eb911fd.png">

**Comment by qiuguian** [comment]
not corrent on phone broser

![2011576489223_ pic](https://user-images.githubusercontent.com/7962840/70896168-2da9b280-202b-11ea-8681-55d4ef2a9b3b.jpg)

**Comment by qiuguian** [comment]
@iapicca  what is the issue going on?  can now fix it?

**Comment by yjbanov** [comment]
Have you tried loading the same video in a plain HTML page using a `<video>` tag without Flutter. Curious if it's a general Safari iOS problem or Flutter-specific.

**Comment by qiuguian** [comment]
@yjbanov  the same video play crrectly,   Are you  use the video_player on phone's browser?  can play crrect?

**Comment by qiuguian** [comment]
the same video play crrectly on mac's browser,  Are you use the video_player on phone's browser? can play crrect?

**Comment by Toolenaar** [comment]
Any update on this? I am facing the same issue. It seems streaming mp4 fails on Safari on iOS. Something todo with the Byte-range headers

**Comment by carman247** [comment]
I am experiencing the same issue. 

I am building a website for someone and the videos uses with video_player_web aren't displaying on Safari (iOS).

I have tried every other browser combination I can and it seems to work fine on all of them.

**Comment by carman247** [comment]
I'm experiencing the same issue.

video_player_web with chewie works fine on every browser combination I've tried so far apart from Safari on iOS.

**Comment by hiroshihorie** [comment]
I have the same issue

**Comment by phide** [comment]
Me too. Does anyone know if other video formats would work?

**Comment by nemanjalugi** [comment]
Same issue

**Comment by yjbanov** [comment]
A work around until this issue is researched and fixed might be to use a `HtmlElementView` and `dart:html` to implement your own `<video>` tag the way Safari understands.

**Comment by rxlabz** [comment]
I have the same issue, and strangely, the work around effectly works.

**Comment by hillelcoren** [comment]
We're also seeing what I think may be the same problem. You can test the app at https://mudeo.app, videos in the app fail to play with a media_err_src_not_supported error but the same videos play correctly in Safari and Chrome when using a video tag. You can click the 'external link' button in the bottom right corner to test the plain HTML video tag.

**Comment by ImDeZz** [comment]
@rxlabz @yjbanov  Could you please provide a working example?

**Comment by rxlabz** [comment]
I created an example : https://rxlabz.github.io/vplayer_safari/#/ the video is played on desktop, but not on safari iOS with plugin. works with the htmlElementView as mentioned by @yjbanov 
src : https://github.com/rxlabz/vplayer_safari cc @ImDeZz

**Comment by martin-matj** [comment]
same issue

**Comment by m-j-g** [comment]
Ran into this as well.  Also,  I couldn't load the video on iOS chrome.  Works fine on both desktop Safari /  desktop Chrome.

**Comment by zeevgrim** [comment]
same issue
:(

**Comment by dannycortesv** [comment]
Same issue, any instructions?

**Comment by kasyrm** [comment]
I also have this issue, any progress on this?

**Comment by faslurrajah** [comment]
I'm getting same kind of problem in ext_video_player plugin...
But I debug it in iOS safari browser and found some exceptions

<img width="996" alt="image" src="https://user-images.githubusercontent.com/30668705/90154000-3f547d00-dda7-11ea-8edd-fc9246581e58.png">


<img width="996" alt="image" src="https://user-images.githubusercontent.com/30668705/90152978-0d8ee680-dda6-11ea-9700-d5e4b725162c.png">

<img width="996" alt="image" src="https://user-images.githubusercontent.com/30668705/90153256-5e9eda80-dda6-11ea-809c-d6f6725b4f6d.png">

<img width="996" alt="image" src="https://user-images.githubusercontent.com/30668705/90153400-8f7f0f80-dda6-11ea-95f7-e726b9dc86de.png">

**Comment by imabhishekkumar** [comment]
same issue here.

**Comment by jing-pei** [comment]
IOS Safari same issue

```
[✓] Flutter (Channel beta, 1.21.0-9.2.pre, on macOS 11.0 20A5364e, locale
    zh-Hans-CN)
    • Flutter version 1.21.0-9.2.pre at /Users/angus/flutter
    • Framework revision 81a45ec2e5 (3 weeks ago), 2020-08-27 14:14:33 -0700
    • Engine revision 20a9531835
    • Dart version 2.10.0 (build 2.10.0-7.3.beta)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/angus/Library/Android/sdk
    • Platform android-R, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.0.beta.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 43.0.3
    • Dart plugin version 193.6015.53

[✓] VS Code (version 1.49.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (3 available)
    • iPhone 8 (mobile) • 7F336652-C2C2-4320-800D-F0C53303503F • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • Web Server (web)  • web-server                           • web-javascript
      • Flutter Tools
    • Chrome (web)      • chrome                               • web-javascript
      • Google Chrome 85.0.4183.102

• No issues found!
```

**Comment by rpk98c** [comment]
I'm experiencing the same issue.

video_player_web  works fine on every browser combination I've tried so far apart from iOS.

**Comment by yaizudamashii** [comment]
For me the videos don't play on Safari on PC either.  I'm using `video_player: ^0.11.1+2` as `video_player_web` is delisted.  The videos play fine on Chrome and Opera.

**Comment by dannycortesv** [comment]
Any solution for this? It's very important to support web mobile for this plugin

**Comment by Leffe108** [comment]
@rxlabz provided a workaround earlier in this thread/issue. I use in production for a hobbyist project and seems to work.

A proper fix in the plugin would still be welcome but I think a Mac is probably necessary to debug/fix the problem and I don’t have that.

**Comment by deepikaaupadhyaya** [comment]
I have a requirement where I need control over the video and show overlay and add music on my video.  I am not able to play in iPhone safari.

**Comment by amirmhmdi** [comment]
I have the same problem ! dose any one find solution for this issue ?

**Comment by amirmhmdi** [comment]
I fix this bug by editing main.dart.js
after build web
open main.dart.js in build/web directory and in video element set q.autoplay = 1 

`function (a) {
            var s, r = this, q = document.createElement("video")
            q.src = r.b
            q.autoplay = 1
            q.controls = !1
            s = q.style
            s.border = "none"
            r.d = q
            q.setAttribute("playsinline", "true")
            $.N5().alE("videoPlayer-" + r.c, new F.aIk(r))
            q = r.d
            q.toString
            s = t.d7.c
            W.c3(q, "canplay", new F.aIl(r), !1, s)
            q = r.d
            q.toString
            W.c3(q, "error", new F.aIm(r), !1, s)
            q = r.d
            q.toString
            W.c3(q, "ended", new F.aIn(r), !1, s)
        },`

**Comment by dannycortesv** [comment]
@amirmhmdi thanks for your solution it works for me too, it is awesome thank uuu!!

**Comment by ferozstk** [comment]
Same issue here

**Comment by WD-J** [comment]
Bumping. Solution doesn't work for me. Chrome on Mac works, so suspecting that it's only an issue related to Safari.

**Comment by abyesilyurt** [comment]
@amirmhmdi's solution worked for me, with some additional changes below:
```
q.autoplay=1
q.controls=1
q.muted=1
```

**Comment by sohailb108** [comment]
> I fix this bug by editing main.dart.js
> after build web
> open main.dart.js in build/web directory and in video element set q.autoplay = 1
> 
> `function (a) { var s, r = this, q = document.createElement("video") q.src = r.b q.autoplay = 1 q.controls = !1 s = q.style s.border = "none" r.d = q q.setAttribute("playsinline", "true") $.N5().alE("videoPlayer-" + r.c, new F.aIk(r)) q = r.d q.toString s = t.d7.c W.c3(q, "canplay", new F.aIl(r), !1, s) q = r.d q.toString W.c3(q, "error", new F.aIm(r), !1, s) q = r.d q.toString W.c3(q, "ended", new F.aIn(r), !1, s) },`

Thanks, that solved my problem.
Is there anyway to tell the developers to fix it, so we don't have to edit the built files every time?

**Comment by Leffe108** [comment]
> Thanks, that solved my problem.
> Is there anyway to tell the developers to fix it, so we don't have to edit the built files every time?

Give OP a thumb up to give this issue higher attention from devs.

**Comment by SebastianKumor** [comment]
Still having this problem

**Comment by TahaTesser** [comment]
I can reproduce it using the [official example](https://github.com/flutter/plugins/tree/master/packages/video_player/video_player/example) 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 2.1.0-13.0.pre.322, on macOS 11.2.3 20D91
    darwin-x64, locale en-GB)
    • Flutter version 2.1.0-13.0.pre.322 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision e93590474d (4 hours ago), 2021-03-30 02:14:03 -0400
    • Engine revision dbcbf69a57
    • Dart version 2.13.0 (build 2.13.0-175.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API
      30) (emulator)
    • macOS (desktop)         • macos         • darwin-x64     • macOS 11.2.3
      20D91 darwin-x64
    • Chrome (web)            • chrome        • web-javascript • Google Chrome
      89.0.4389.90

• No issues found!
```
	
</details>

**Comment by git9am** [comment]
Another workaround if you don't want to touch the built main.dart.js.

1. Download the video_player_web project from https://github.com/flutter/plugins/tree/master/packages/video_player/video_player_web
2. Locate initialize in video_player_web.dart and update those "autoplay" values to true.
3. Adds to the pubspec.yaml:

```
dependency_overrides:
  video_player_web:
    path: ./video_player_web
```

**Comment by Abdelrahman-Sousa** [comment]
Still having the issue. Unfortunately @amirmhmdi 's fix does not work for me.

**Comment by chntu** [comment]
Same. The solution doesn't work for me either. Does anyone have other packages that work?

**Comment by Abdelrahman-Sousa** [comment]
> Still having the issue. Unfortunately @amirmhmdi 's fix does not work for me.

After much trial and error, renaming the video somehow solved it.
My videos on Firebase did not end with .mp4, but still worked perfectly on android and pc.
Renaming the video to end with .mp4 combined with Amir's fix did it for me.

**Comment by bortka** [comment]
Hi, we are having problems getting this to work for flutter web. The player never initializes and just hangs. None of the web solutions mentioned here work with the current version of flutter.

**Comment by ditman** [comment]
Hello all! I'm sorry I haven't commented here before (I thought I had!). This issue has grown for a while, and from what I've read, I think we're actually having several issues here.

I'm going to try and explain the issues I think we have, and maybe steps to verify each of them.

1. **The video file is not compatible with the browser.** Not all browsers are able to open the same video formats. This is documented [in the README](https://pub.dev/packages/video_player_web#supported-formats).
2. **The video will not autostart.** Autostart/autoplay is [severely limited](https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide) in browsers. As a very simplified heuristic, only videos without sound (or that are muted) are allowed to auto-start without user interaction.

### How to know what's affecting me?

Test your use case in a raw HTML page:

1. You can create a very simple HTML page with a `video` tag that attempts to open your video. [See examples](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#examples). If your video plays back in an HTML page, it should play in Flutter Web (the web version of the plugin is using a standard video tag). If your video doesn't play in a specific browser, it might need to be re-encoded for that browser. This normally cannot be done client-side, and you might need to have several versions of the same video for it to work across all browsers.
2. Try to start your video muted (`.setVolume(0)`) before calling `play`. Normally there's no need to add the `autoplay` attribute by hand; calling `play` as soon as the controller is ready, should be the same as having `autoplay`. You can also use the suggestion in "1" to create a raw HTML page with an auto-playing video tag, and see if it works there or not.

If after testing the above, you're finding it's a different issue, please, **do create a separate issue for it**, so it doesn't get mixed up/lost within the comments of this one!

### How to report more info

The amount of video formats and browser combinations can be enormous. Ideally, when reporting a +1, please add:

* What is the **OS and Browser** where the issue is happening? (Try to be as specific as possible with the version info.)
* **What have you tried?** Can you include some samples from your testing?
  * **Super helpful:** can you include the video (or a part of it) that is giving you trouble?
  * A small Flutter Web repro case with code similar to what your page needs to do (are you opening a video after a click? immediately after navigating to a new tab?)
  * (Maybe show us the HTML test page you built in the step above too where the issue reproduces?)
* Are there **any errors in the Chrome DevTools?**
  * Check the **JS Console**, and the **Network panels**.

_Thanks for reading the wall of text above! Let's get the video player working great on the web!_

**Comment by WSDDi** [comment]
flutter web项目 [video_player]视频无法在iOS浏览器中打开播放

**Comment by Leffe108** [comment]
In response to [previous comment](https://github.com/flutter/flutter/issues/47030#issuecomment-841419012), here is a reproducable example that fulfills:
* Flutter video_player do not play on iOS Safari
* HTML5 do play the video on iOS Safari 
* The video plays fine in both flutter video_player and html5 on other tested platforms (Chrome on Android, Chrome on Windows and Firefox on windows). 

The two iOS devices that I have tested on are:
* iPhone SE (model A1723) with iOS 14.4.2
* iPad pro (model A2230) with iOS 14.5.1

## Steps to Reproduce

1. Run `flutter create bug_video`.
2. Update the files as follows: 

In pubspec.yaml add these two dependencies:
```
  video_player: 2.1.1
  url_launcher: 6.0.4
```

In web/index.html update base to desired base url for hosting location.

<details>
  <summary>lib/main.dart</summary>

```dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';
import 'package:video_player/video_player.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Video demo'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _playIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            if (kIsWeb)
              ElevatedButton(
                  onPressed: () async {
                    // Go to html5.html on same host as index.html is served at
                    Uri uri = Uri.base;
                    String path = uri.path.replaceAll('index.html', '');
                    if (!path.endsWith('/')) path += '/';
                    path += 'html5.html';
                    uri = uri.replace(path: path, fragment: null);
                    if (await canLaunch(uri.toString())) {
                      launch(uri.toString());
                    }
                  },
                  child: Text('Go to HTML player')),
            Text('W'),
            MyVideoPlayer(
                key: Key('video $_playIndex'),
                url:
                    'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4'),
            Text('Video is CC-BY-NC-SA 2.5 from Svenskt Teckenspråkslexikon'),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.play_arrow),
        onPressed: () {
          setState(() {
            this._playIndex += 1;
          });
        },
      ),
    );
  }
}

class MyVideoPlayer extends StatefulWidget {
  final String url;

  const MyVideoPlayer({Key? key, required this.url}) : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  late VideoPlayerController _controller;

  @override
  void initState() {
    _initController();
    super.initState();
  }

  void _initController() async {
    _controller = VideoPlayerController.network(widget.url);
    await _controller.initialize();
    _controller.setVolume(0);
    setState(() {
      _controller.play();
    });
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return AspectRatio(
      aspectRatio: _controller.value.isInitialized
          ? _controller.value.aspectRatio
          : 16 / 9,
      child: _controller.value.isInitialized
          ? VideoPlayer(_controller)
          : Container(
              color: Colors.black,
              child: Center(
                child: CircularProgressIndicator(),
              ),
            ),
    );
  }
}
```

</details>

<details>
  <summary>web/html5.html</summary>

```dart
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <title>bug_video html5 version</title>
  <style>
    body {
      margin: 0;
      text-align: center;
    }
  </style>
</head>
<body>
  <p>W</p>
  <video 
    autoplay
    playsinline
    controls
    src="https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4"
  ></video>
  <p>Video is CC-BY-NC-SA 2.5 from Svenskt Teckenspråkslexikon</p>
</body>
</html>
```

</details>


3. Deploy web build or use [my deployment](https://tspquiz.se/bug-47030/) (`flutter build web`) and open in Safari on iPhone or iPad
4. Now the video should play 
5. To open the HTML5 video player press on the button above the video or go to https://tspquiz.se/bug-47030/html5.html

**Expected results:** 
I expect the video to play after it has been loaded on iOS devices. 

**Actual results:** 
It does not play in flutter video_player.

<details>
  <summary>Logs</summary>



```
Analyzing bug_video...
No issues found! (ran in 5.8s)
```



```
[✓] Flutter (Channel stable, 2.2.0, on Microsoft Windows [Version 10.0.19041.985], locale sv-SE)
    • Flutter version 2.2.0 at C:\Users\Leif\Programmering\Libraries\flutter
    • Framework revision b22742018b (5 days ago), 2021-05-14 19:12:57 -0700
    • Engine revision a9d88a4d18
    • Dart version 2.13.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Leif\Programmering\Android\android-sdks
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code, 32-bit edition (version 1.56.2)
    • VS Code at C:\Program Files (x86)\Microsoft VS Code
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 90.0.4430.212
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 90.0.818.62

• No issues found!
```

</details>

**Comment by ditman** [comment]
Excellent report, thanks @Leffe108!

**Comment by tsdevelopment** [comment]
Hi,

I had the same issue and needed the video functionality for a current project. So I added the following line in the `video_player_web-2.0.0/lib/video_player_web.dart` package (line 182):

```dart
// Allows Safari iOS to play the video inline
videoElement.setAttribute('playsinline', 'true');
videoElement.setAttribute('autoplay', 'false'); // <-- Added this line. Safari seems to need this tag although it's false
```

Works good with the latest Safari on IPhone. Maybe it would good to influence such attributes from outside...

**Comment by felangel** [comment]
I opened https://github.com/flutter/plugins/pull/3985 to disable autoplay as described above

**Comment by mahanabd** [comment]
@abyesilyurt 
I change main.dart.js like you but the problem not solved. (in Html & canvas kit release)

**Comment by Leffe108** [comment]
> 
> 
> > I opened [flutter/plugins#3985](https://github.com/flutter/plugins/pull/3985) to disable autoplay as described above
> 
> This has been merged and published: https://pub.dev/packages/video_player_web/changelog#201

Is there a way to test this?

I tried to add `video_player_web: ^2.0.1` as a dependency and re-build my demo but it doesn't fix the problem. When I inspect the video tag on a desktop browser (firefox) in windows the video tag does not get autoplay set to false. I don't know however if it is supposed to do that on desktop browsers too or only iPhone|iPod|iPad|MacIntel?

**Comment by ditman** [comment]
@Leffe108 try deleting your `pubspec.lock` to ensure that you're getting the latest version of `video_player_web` (or check inside pubspec.lock what version of `video_player_web` you're pub-getting).

From the code that @felangel wrote, it should be adding `autoplay=false` **unconditionally**. [Here](https://github.com/flutter/plugins/blob/master/packages/video_player/video_player_web/lib/video_player_web.dart#L191-L192).

**Comment by Leffe108** [comment]
@ditman [My test case](https://github.com/flutter/flutter/issues/47030#issuecomment-845318803) now works using video_player_web 2.0.1.

I accidentally deployed ./web instead of ./build/web and forgot to update base url so it still used the old code. 🤦‍♀️

**Comment by ditman** [comment]
That's great news @Leffe108! Let's see if others in the thread experience an improvement! (I hope all who were adding autoplay=false to the compiled JS directly would see the improvement as well!)

**Comment by osho-chawla** [comment]
Hi, even after using `video_player_web: ^2.0.1` video don't play in Safari/Chrome on iOS. Is anyone here still experiencing this?

**Comment by ditman** [comment]
@osho-chawla please, take a look at this: https://github.com/flutter/flutter/issues/47030#issuecomment-841419012

**Comment by sxmeer-ahmed** [comment]
My Problem is little different,
What I did : I have implemented video_player in flutter web for background in stack.
Problem : Video Player is not playing video after load.
Browser am using : Chrome
Extras : Video is working fine and all but I have reload my page inside my website(not chrome button).
Wht should I do now?
Website Link : https://sameer-ahmed-portfolio.web.app/

**Comment by Leffe108** [comment]
> My Problem is little different,
> What I did : I have implemented video_player in flutter web for background in stack.
> Problem : Video Player is not playing video after load.
> Browser am using : Chrome
> Extras : Video is working fine and all but I have reload my page inside my website(not chrome button).
> Wht should I do now?
> Website Link : https://sameer-ahmed-portfolio.web.app/

I would bet that the video plays without user interaction and therefore gets blocked by chrome which I IIRC is stated in dev console when it happens. I had that issue when I implemented a share quiz function that deep link straight into a question. I solved it with an extra UI dialog informing user they got challenged and ask if they accept with yes/no button obtaining user interaction prior to playing video.

**Comment by sxmeer-ahmed** [comment]
Thanks @Leffe108,
Now, I am thinking to implement a welcome screen before which will forward to first screen and try to get a call from there :)

**Comment by mrinaljain** [comment]
Following is my sample code for safari/ios, I have turned autoplay off and volume to 0 because safari does not support autoplay  of unmuted video
flick_video_player can play in Mac safari normally by platform web, but when I play in phone safari, it doesn't work
**### only audio can be heard but there is no video visible.**
```
flickManager = FlickManager(
          videoPlayerController: VideoPlayerController.network(
              "https://flutter.github.io/assets-for-api-docs/assets/videos/bee.mp4")
          autoPlay: false);
      flickManager?.flickControlManager?.mute();
      flickManager?.flickControlManager?.setVolume(0);
```

an issue regarding the video_player_web was resolved here but the problem still persists in the flick video player.

**Comment by osho-chawla** [comment]
@sxmeer-ahmed I had the same problem. Auto play is restricted by browsers.
One option as stated is to make a dialog for user interaction. 
Other option is to start video on mute, and let users know their video is on mute. So just before videoController.play(), set the volume to 0. I think this way is more elegant than other one.

**Comment by kira2015** [comment]
May I ask, do you have the final solution? I also encountered this problem, very anxious

**Comment by zambetpentru** [comment]
In case it digs someone out of a hole while the main issue is being resolved, this is working well for me.

`import 'dart:html';
import 'dart:ui' as ui;
import 'package:flutter/material.dart';

///Working well in Windows, Linux, MacOS, iOS and Android Mobile Web / PWA
class WebVideoElement extends StatefulWidget {
const WebVideoElement(this.src);

final String src;
final double startAt = 0;
final bool autoplay = false;
final bool controls = true;

@override
_WebVideoElementState createState() => _WebVideoElementState();
}

class _WebVideoElementState extends State {
@override
void initState() {
super.initState();

String? URL = 'http://www.website.com/pathtovideo' + '#t=${widget.startAt}';
// Do not remove the below comment - Fix for missing ui.platformViewRegistry in dart.ui
// ignore: undefined_prefixed_name
ui.platformViewRegistry.registerViewFactory(widget.src, (int viewId) {
  //https: //api.flutter.dev/flutter/dart-html/VideoElement-class.html
  final video = VideoElement()
    ..src = URL
    ..autoplay = widget.autoplay
    ..controls = widget.controls
    ..style.border = 'none'
    ..style.borderColor = 'red'
    ..style.height = '100%'
    ..style.width = '100%';

  // Allows Safari iOS to play the video inline
  video.setAttribute('playsinline', 'true');

  return video;
});
}

@override
Widget build(BuildContext context) {
return HtmlElementView(viewType: widget.src);
}
}`

**Comment by Leffe108** [comment]
@zambetpentru There was a fix [implemented in video_player_web 2.0.1](https://github.com/flutter/flutter/issues/47030#issuecomment-852564661) which solved the problem for some. To use it add `video_player_web: ^2.0.1` to your pubspec.yaml and rebuild and re-deploy.

If this does not solve the problem in your app with your video source, then please take a look at this: https://github.com/flutter/flutter/issues/47030#issuecomment-841419012 on how to report the specifics of your case.

**Comment by amirmhmdi** [comment]
on flutter 2.5.1 video player 2.2.4 
video play on android iOS and chrome and safari Mac  ,but safari iOS doesn't play.

**Comment by a-eniou-pvotal** [comment]
Autoplay is not working for me on Safari Mac  Version 15.4 
So I wonder is this the right implementation for auto-play

>       child: FutureBuilder(
>         future: Future.wait([
>           _initializeVideoFuture,
>           _videoController.setVolume(0),
>           _videoController.setLooping(true),
>           _videoController.play(),
>         ]),


```
import 'package:flutter/material.dart';
import 'package:frontend_framework/frontend_framework.dart';

class AnimatedAsset extends StatefulWidget {
  const AnimatedAsset({
    Key key,
    @required this.asset,
    this.alternativeImage = '',
    @required this.packageName,
    @required this.width,
    @required this.height,
  }) : super(key: key);

  final String asset, alternativeImage, packageName;
  final double width, height;

  @override
  State<AnimatedAsset> createState() => _AnimatedAssetState();
}

class _AnimatedAssetState extends State<AnimatedAsset> {
  VideoPlayerController _videoController;
  ChewieController _chewieController;
  Future<void> _initializeVideoFuture;

  @override
  void initState() {
    super.initState();
    _videoController = VideoPlayerController.asset(
      widget.asset,
      package: widget.packageName,
    );

    _initializeVideoFuture = _videoController.initialize().then(
      (value) async {
        // muted videos can autoplay on chrome
      },
    );
  }

  @override
  void dispose() {
    _videoController.dispose();
    _chewieController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    //video has no auto-play feature on Safari so I shows animated image
    //and macOS has no support for video player yet
    final useAlternativeImage =
        (UniversalPlatform.isDesktop || BrowserDetector().isSafari) &&
            widget.alternativeImage.isNotEmpty;

    if (useAlternativeImage) {
      return Image.asset(
        widget.alternativeImage,
        width: widget.width,
        height: widget.height,
        package: widget.packageName,
        fit: BoxFit.contain,
      );
    }

    return SizedBox(
      width: widget.width,
      height: widget.height,
      child: FutureBuilder(
        future: Future.wait([
          _initializeVideoFuture,
          _videoController.setVolume(0),
          _videoController.setLooping(true),
          _videoController.play(),
        ]),
        builder: (context, snapshot) {
          if (snapshot.connectionState == ConnectionState.done) {
            _chewieController = ChewieController(
              videoPlayerController: _videoController,
              autoPlay: true,
              looping: true,
              showControls: false,
              showControlsOnInitialize: true,
              showOptions: false,
              allowMuting: true,
              autoInitialize: true,
            );

            return Chewie(
              controller: _chewieController,
            );
          }
          return const PyrinasLoadingIndicator();
        },
      ),
    );
  }
}
```

**Comment by Leffe108** [comment]
@a-eniou-pvotal:

Are you trying to auto-play before the user has tapped/interacted with your web-app or after user interaction? I know chrome and likely also safari will block auto-play if you try to play a video before user has interacted with a website/web app. I solved this in my quiz app when receiving a share link to on web show a prompt dialog that asks the user if they want to accept the challenge they received or not. If they accept it, I then auto-play first question video and it works on web.

If your video has audio, try to auto-play muted (after user interaction) to see if that helps.

**Comment by a-eniou-pvotal** [comment]
@Leffe108 so why this sample works fine on safari but not Flutter even I used same HTML content but it didn't work on safari
 https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_video_autoplay

**Comment by amorenew** [comment]
@zambetpentru my issue with auto-play, and I tried your snippet but still no auto-play on Safari mac

**Comment by Leffe108** [comment]
> @zambetpentru my issue with auto-play, and I tried your snippet but still no auto-play on Safari mac

Notice that this issue so far has been about problems with playback in safari on iOS - not safari on mac. Furthermore many people report that it works fine in Safari on mac.

**Comment by ikarelin** [comment]
Nope, I have a problem even on safari on Mac (in flutter web).
When I compiling application to iOS player is working.

**Comment by ikarelin** [comment]
Seems the player is downloading MP4 by pieces. And the link to MP4 is opening normally without video player. Seems to be a bug in flutter video_player plugin.

**Comment by ikarelin** [comment]
<img width="1057" alt="Screenshot 2022-05-12 at 16 53 07" src="https://user-images.githubusercontent.com/1290032/168091331-32ed25c3-7261-443f-bfaf-56e1e68e27f3.png">

**Comment by ikarelin** [comment]
Web console in Safari on Mac.

**Comment by dannycortesv** [comment]
Hello guys, any solution for this? I rendered my project with html mode and it is playing properly on Chrome but not on Safari

**Comment by zambetpentru** [comment]
> Hello guys, any solution for this? I rendered my project with html mode and it is playing properly on Chrome but not on Safari

Hi, did you try my code above? It's still working consistently for me.

**Comment by dannycortesv** [comment]
@zambetpentru i need to use video_player plugin,  it is working well on Chrome but not on Safari.... Strange, before it was working properly...

**Comment by ditman** [comment]
There has been a recent fix with clipped platform views in safari that might have fixed some of the remaining cases on this issue.

If your video player was inside any form of clip-path (like a RRect), Safari wouldn't paint it before this landed:

* https://github.com/flutter/engine/pull/35948

(Just letting buddies on this venerable issue know!)

**Comment by deepak786** [comment]
> https://github.com/flutter/engine/pull/35948
Waiting for this to land in Stable version.

**Comment by aytunch** [comment]
@ditman We face this issue on Safari web in CanvasKit still. Is it not on Stable yet? When can we expect it to land? Thanks for all your efforts.

**Comment by deepak786** [comment]
> https://github.com/flutter/engine/pull/35948
This fix is landed in Stable 3.3.10. 
Still the ClipRRect is not showing the full content on Safari.
I'm using the ClipRRect around the Html Video Element. And the player is not showing the full video content using the ClipRRect.

**Comment by okiabrian123** [comment]
im use Canvaskit, sometimes video player fail load video from url in safari browser on ios simulator, but its always worked on google chrome, and rare fail in mac os

**Comment by razfazz** [comment]
kind of sad that this issue is still not fixed. maybe exposing video attributes could help to apply the needed settings.

https://stackoverflow.com/a/72658854/5525799

**Comment by zacherygentry** [comment]
bump

**Comment by zacherygentry** [comment]
Using media_kit seems to work
https://pub.dev/packages/media_kit

**Comment by PawlikMichal25** [comment]
@TahaTesser almost 4 years and we don't have such a basic thing like playing videos everywhere on Flutter Web? Seriously?

**Comment by PawlikMichal25** [comment]
@TahaTesser @yjbanov 
Almost 4 years and we don't have such a basic thing like playing videos everywhere on Flutter Web? Seriously?

**Comment by PawlikMichal25** [comment]
@yjbanov @TahaTesser @ditman 
It's quite disappointing to see such basic thing as videos still not working in Flutter Web 😕

**Comment by PawlikMichal25** [comment]
Even running official example from https://pub.dev/packages/video_player#example on Safari doesn't work. The video is just not showing. 

Env:
- flutter: 3.13.7
- video_player 2.7.2
- iOS 16.7.1

**Comment by ditman** [comment]
I've published a fix for the issue that @PawlikMichal25 reported:

* https://github.com/flutter/flutter/issues/137023

In `video_player_web: ^2.1.2` (which should be automatically be picked up by your apps as you upgrade your dependencies).

Here's a sample app: https://dit-videoplayer-tests.web.app (may not work if I've repurposed the domain for anything!)

**Comment by dev-weenit** [comment]
I am facing a similar issue with `video_player: 2.8.2` on Edge browser Android 14.

In the case of iOS using the latest flutter version (3.13.9 -> 3.16.8) worked for me.

**Comment by ZephyrRaine** [comment]
Hello, I'm currently having the same issue. This is what I have when printing some values from the controller : 

```
[Log] has error: true (dart_sdk.js, line 31999)  
[Log] error description: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission. (dart_sdk.js, line 31999)
```
It works on all other platforms and is used like so: 

```Dart
  void _initializeVideoPlayer() {
    _controller = VideoPlayerController.networkUrl(Uri.parse(widget.videoUrl))
      ..initialize().then((_) async {
        if (!mounted) {
          return;
        }
        setState(() {
          _isInitialized = true;
        });
        await _controller.play();

        _videoListener = () {
          if (_controller.value.position >= _controller.value.duration &&
              widget.onVideoEnded != null) {

            widget.onVideoEnded!();
          }
        };

        _controller.addListener(_videoListener!);
      });
  }
```
Please let me know if you have any idea :)

**Comment by amber-lau** [comment]
@ZephyrRaine I am having issue playing video in iOS browser too, do you have a solution?

**Comment by ZephyrRaine** [comment]
@amber-lau Unfortunately I do not as of now, I'll be looking into dedicated flutter packages shortly, will keep you posted.

**Comment by amber-lau** [comment]
**Issue**
I am using PageView to load the VideoPlayer.
In Safari, it successfully play video with sound on until the 5th video.
In Chrome, it cannot play even one video with sound on.
**Summary of findings** 
I tried to add ..setVolume(0.0) to controller. It then works fine. Also, tried another way, pause the video when scrolling down the page. It is also working. I think it's the autoplay policy making it forever loading. 
The strange thing is...it is only affecting iPhone Safari and Chrome. Is it possible to avoid this on iPhone Safari and Chrome? Actually users are already interacting with the website when scrolling down the PageView, so it should not be blocked by the policy.

**Comment by ditman** [comment]
@amber-lau I think you're encountering different autoplay restrictions from the different browsers, see this:
https://developer.mozilla.org/en-US/docs/Web/Media/Guides/Autoplay

Especifically the "Autoplay availability" section.

**Comment by amber-lau** [comment]
@ditman Thanks for replying. 
In my case, user clicks on a small muted VideoPlayer or click on navigation tab, then they will be redirect to the page with autoplay-video PageView. So, this situation meet the requirement `The user has interacted with the domain`. However, it is not playing even one video in iOS Chrome; while Safari can play up to first 4 videos, then issue starts happening.
Android Chrome, Android Firefox and iOS Firefox are working fine.
**Test 1**
I tried to play a video with HTMLVideoElement, user clicks on navigation tab before displaying the page with video. Both Safari and iOS Chrome are allowed to play video with sound.
**Test 2**
I also tried to make the first video in PageView muted, after user tries to unmute the video, the following videos will autoplay with sound. However, the video with sound still not playing even it is the previous video is unmute by user.

**Comment by ditman** [comment]
@amber-lau can you share a reproduction code of your problem? If your videos are off-screen (you need to scroll to see them), they will **not** be added to the DOM of the page until they scroll close enough to the viewport (and for the browser, they will be "new" in the page, so the interaction probably doesn't apply to them).

As a very inefficient hack, could you try replacing the contents of your `PageView` (which does proper scrolling) with a `SingleChildScrollView + Column`? That should force all elements to render at once, and could help validate if my theory is correct?

**Comment by amber-lau** [comment]
@ditman My code is here.
https://github.com/flutter/flutter/issues/171981

I tried SingleChildScrollView + Column. I loaded 6 players when the page initialized and load more on scroll, those initial players can play, but those from load more cannot. 
Seems it is the same situation with the PageView...

**Comment by mdebbar** [comment]
@amber-lau could you try the same thing in a plain HTML+JS+CSS app? (a jsfiddle might be the easiest way to play with it).

This will tell us if it's a flutter problem or a browser problem. Make sure you do the same behavior in the jsfiddle: load 6 players initially, then load more later.

**Comment by amber-lau** [comment]
@mdebbar Even worse, I can't autoplay a single video even it is muted.
